index.js

```diff
@@ -1,60 +1,141 @@
+  if (name) {
+    Listing.find({ Location: name }, function (err, listing) {
+      res.status(200).json(listing);
+    });
+  } else {
+    Listing.find(function (err, listings) {
+      if (err) {
+        console.log(err);
+      } else {
+        res.status(200).json(listings);
+        console.log();
       }
-    ]
+    });
   }
-}
+});
+
+//Create new Listing
+listingRoutes.route("/").post(function (req, res) {
+  let listing = new Listing(req.body);
+  listing
+    .save()
+    .then((listing) => {
+      res.status(201).json(listing);
+    })
+    .catch((err) => {
+      res.status(400).send(err);
+    });
+});
 
-app.use('/', express.static(__dirname + '/swagger'));
-app.use('/swagger', swaggerUi.serve, swaggerUi.setup(null, options));
-app.use('/', routes)
+//GET Listing by id
+listingRoutes.route("/:id").get(function (req, res, next) {
+  let id = req.params.id;
+  Listing.findById(id, function (err, listing) {
+    res.status(200).json(listing);
+  });
+});
 
+//DELETE Listing by id
+listingRoutes.route("/:id").delete((req, res) => {
+  Listing.findByIdAndDelete(req.params.id)
+    .then(() => res.status(200).send("Deleted"))
+    .catch((err) => res.status(400).json("Error: " + err));
+});
 
-// Start up the database, then the server and begin listen to requests
-if(process.env.NODE_ENV != "test") {
-  db.connectDb().then(() => {
-    const listener = app.listen(port, () => {
-      console.info(`Server is listening on port ${listener.address().port}.`);
-    })
-  }).catch((error) => {
-    console.error(error)
-  })
-}
+listingRoutes.route("/:id").put(function (req, res) {
+  Listing.findById(req.params.id, function (err, listing) {
+    if (!listing) res.status(404).send(err);
+    else {
+      listing.Address = req.body.Address;
+      listing.Location = req.body.Location;
+      listing.Price = req.body.Price;
+      listing.MonthlyFee = req.body.MonthlyFee;
+      listing.Type = req.body.Type;
+      listing.Coordinate.Longitude = req.body.Coordinate.Longitude;
+      listing.Coordinate.Latitude = req.body.Coordinate.Latitude;
+
+      listing
+        .save()
+        .then((listing) => {
+          res.status(200).json(listing);
+        })
+        .catch((err) => {
+          next(err);
+        });
+    }
+  });
+});
+
+//PUT updates listing
+listingRoutes.route("/:id").put(function (req, res) {
+  Listing.findById(req.params.id, function (err, listing) {
+    if (!listing) res.status(404).send(err);
+    else {
+      listing.Address = req.body.Address;
+      listing.Location = req.body.Location;
+      listing.Price = req.body.Price;
+      listing.MonthlyFee = req.body.MonthlyFee;
+      listing.Type = req.body.Type;
+      listing.Coordinate.Longitude = req.body.Coordinate.Longitude;
+      listing.Coordinate.Latitude = req.body.Coordinate.Latitude;
+
+      listing
+        .save()
+        .then((listing) => {
+          res.status(200).json(listing);
+        })
+        .catch((err) => {
+          next(err);
```

Comment:
beautiful, brings a goddamn tear to my eye!

fast du glömde next i signaturen:
```javascript
listingRoutes.route("/:id").put(function (req, res, next) {
```